src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -648,6 +648,11 @@ FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::FFN(
     this->network.push_back(boost::apply_visitor(copyVisitor,
         network.network[i]));
   }
+
+  for (size_t i = 0; i < this->network.size(); ++i)
```

Comment:
Consider using `for (const auto& net : network) { ... }` instead of old school and error prone for-loop